From f10e25e99907adc34af55efb6b704896c42065d4 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Thu, 14 Dec 2023 10:46:09 +0100
Subject: [PATCH 0212/1695] dts: stm32: populate UART nodes in stm32mp251 SoC
 device tree file

Populate all UART/USART nodes present in this device tree file.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git f6139ede662214460c03b9d47ac344a04b5565c1 dts: stm32: populate UART nodes in stm32mp251 SoC device tree file]
Change-Id: I5221971da5715066a92f44adcbcc3efc9c6a3136
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 858aad7e5..a09da0a1b 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -123,22 +123,34 @@
 			#size-cells = <1>;
 
 			usart2: serial@400e0000 {
+				compatible = "st,stm32h7-uart";
 				reg = <0x400e0000 0x400>;
+				interrupts = <GIC_SPI 115 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_KER_USART2>;
 				status = "disabled";
 			};
 
 			usart3: serial@400f0000 {
+				compatible = "st,stm32h7-uart";
 				reg = <0x400f0000 0x400>;
+				interrupts = <GIC_SPI 116 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_KER_USART3>;
 				status = "disabled";
 			};
 
 			uart4: serial@40100000 {
+				compatible = "st,stm32h7-uart";
 				reg = <0x40100000 0x400>;
+				interrupts = <GIC_SPI 126 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_KER_UART4>;
 				status = "disabled";
 			};
 
 			uart5: serial@40110000 {
+				compatible = "st,stm32h7-uart";
 				reg = <0x40110000 0x400>;
+				interrupts = <GIC_SPI 127 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_KER_UART5>;
 				status = "disabled";
 			};
 
@@ -153,27 +165,42 @@
 			};
 
 			usart6: serial@40220000 {
+				compatible = "st,stm32h7-uart";
 				reg = <0x40220000 0x400>;
+				interrupts = <GIC_SPI 136 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_KER_USART6>;
 				status = "disabled";
 			};
 
 			uart9: serial@402c0000 {
+				compatible = "st,stm32h7-uart";
 				reg = <0x402c0000 0x400>;
+				interrupts = <GIC_SPI 150 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_KER_UART9>;
 				status = "disabled";
 			};
 
 			usart1: serial@40330000 {
+				compatible = "st,stm32h7-uart";
 				reg = <0x40330000 0x400>;
+				interrupts = <GIC_SPI 114 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_KER_USART1>;
 				status = "disabled";
 			};
 
 			uart7: serial@40370000 {
+				compatible = "st,stm32h7-uart";
 				reg = <0x40370000 0x400>;
+				interrupts = <GIC_SPI 148 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_KER_UART7>;
 				status = "disabled";
 			};
 
 			uart8: serial@40380000 {
+				compatible = "st,stm32h7-uart";
 				reg = <0x40380000 0x400>;
+				interrupts = <GIC_SPI 149 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_KER_UART8>;
 				status = "disabled";
 			};
 
-- 
2.47.2

